lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,30 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        default_address.merge(top_level_attributes).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash
+          .merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+          .delete_if { |_, v| v.nil? }
+      end
+
+      def default_address
+        if at_least_one_address?
+          record_hash['addresses'][0].slice(*ADDRESS_ATTRIBUTES)
+        else
+          {}
+        end
+      end
+
+      def at_least_one_address?
+        record_hash.dig('addresses').present? && record_hash.dig('addresses')[0].present?
```

Comment:
No need for the dig since it's one level deep access. If you were doing something like `record_hash.dig('addresses', 'something_else'), then dig is necessary.

```ruby 
record_hash['addresses'].present? && record_hash['addresses'].first.present?